spec/rubocop/cop/sorbet/obsolete_strict_memoization_spec.rb

```diff
@@ -0,0 +1,197 @@
+      end
+    RUBY
+  end
+
+  describe "the obsolete memoization pattern" do
+    it "registers an offence and autocorrects" do
+      expect_offense(<<~RUBY)
+        def foo
+          @foo = T.let(@foo, T.nilable(Foo))
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ This two-stage workaround for memoization in `#typed: strict` files is no longer necessary. See https://sorbet.org/docs/type-assertions#put-type-assertions-behind-memoization.
+          @foo ||= Foo.new
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+        def foo
+          @foo ||= T.let(Foo.new, T.nilable(Foo))
+        end
+      RUBY
+    end
+
+    describe "with a complex type" do
+      it "registers an offence and autocorrects" do
+        expect_offense(<<~RUBY)
+          def client_info_hash
+            @client_info_hash = T.let(@client_info_hash, T.nilable(T::Hash[Symbol, T.untyped]))
+            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ This two-stage workaround for memoization in `#typed: strict` files is no longer necessary. See https://sorbet.org/docs/type-assertions#put-type-assertions-behind-memoization.
+            @client_info_hash ||= client_info.to_hash
+          end
+        RUBY
+
+        expect_correction(<<~RUBY)
+          def client_info_hash
+            @client_info_hash ||= T.let(client_info.to_hash, T.nilable(T::Hash[Symbol, T.untyped]))
+          end
+        RUBY
+      end
+    end
+
+    describe "with a long initialization expression" do
+      it "registers an offence and autocorrects into a multiline expression" do
+        expect_offense(<<~RUBY)
+          def foo
+            @foo = T.let(@foo, T.nilable(SomeReallyLongTypeName______________________________________))
+            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ This two-stage workaround for memoization in `#typed: strict` files is no longer necessary. See https://sorbet.org/docs/type-assertions#put-type-assertions-behind-memoization.
+            @foo ||= some_really_long_initialization_expression______________________________________
+          end
+        RUBY
+
+        expect_correction(<<~RUBY)
+          def foo
+            @foo ||= T.let(
+              some_really_long_initialization_expression______________________________________,
+              T.nilable(SomeReallyLongTypeName______________________________________),
+            )
+          end
+        RUBY
+
+        autocorrected_source = autocorrect_source(<<~RUBY)
+          def foo
+            @foo = T.let(@foo, T.nilable(SomeReallyLongTypeName______________________________________))
+            @foo ||= some_really_long_initialization_expression______________________________________
+          end
+        RUBY
+
+        longest_line = autocorrected_source.lines.max_by(&:length)
+        expect(longest_line.length).to(be <= 120) # FIXME: Unhardcode this 120
```

Comment:
I assume 120 is meant to match the `Layout/LineLength` `Max`? I think the "correct" way to do this would be to override the config in this test to specify that value explicitly.

e.g. https://github.com/rubocop/rubocop/blob/d69fcd0964a441e085a199deb0f55d2e87731a0d/spec/rubocop/cop/style/safe_navigation_spec.rb#L958-L965
e.g. https://github.com/rubocop/rubocop/blob/d69fcd0964a441e085a199deb0f55d2e87731a0d/spec/rubocop/cop/layout/redundant_line_break_spec.rb#L3-L9

Reply:
I think this might have gotten lost among the previous reviews.

Reply:
Indeed it was, thanks for the nudge and the pointer. This approach worked great!